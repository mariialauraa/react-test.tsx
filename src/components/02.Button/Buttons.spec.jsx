/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react"
import { Buttons } from "."
import userEvent from "@testing-library/user-event"

describe('<Buttons />', () => {

    it('should render the button with the text "Load more"', () => {
        render(<Buttons text="Load more" />)
        expect.assertions(1)

        const button = screen.getByRole('button', { name: /load more/i })
        expect(button).toBeInTheDocument()
        // expect(button).toHaveAttribute('class', 'button')
    })

    it('should call function on button click', async () => {
        // cria um mock para simular a função
        const fn = jest.fn()

        render(<Buttons text="Load more" onClick={fn} />)

        const button = screen.getByRole('button', { name: /load more/i })

        // aqui exige que a função seja assíncrona
        await userEvent.click(button);
        expect(fn).toHaveBeenCalledTimes(1);

        // aqui não
        fireEvent.click(button)
        fireEvent.click(button)
        expect(fn).toHaveBeenCalledTimes(3)
    })

    it('should be disabled when disabled is true', () => {

        render(<Buttons text="Load more" disabled={true} />)

        const button = screen.getByRole('button', { name: /load more/i })

        expect(button).toBeDisabled()
    })

    it('should be enabled when disabled is false', () => {

        render(<Buttons text="Load more" disabled={false} />)

        const button = screen.getByRole('button', { name: /load more/i })

        expect(button).toBeEnabled()
    })
})