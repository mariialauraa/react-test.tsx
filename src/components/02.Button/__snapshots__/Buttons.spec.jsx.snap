// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Buttons /> should match snapshot 1`] = `
<button
  class="button"
>
  Load more
</button>
`;
